lib/store_web/live/product_live/form_component.ex

```diff
@@ -37,11 +37,32 @@ defmodule StoreWeb.ProductLive.FormComponent do
       {:ok, _product} ->
         {:noreply,
          socket
-         |> put_flash(:success, "Product purchased successfully!")
-         |> push_redirect(to: Routes.cart_index_path(socket, :index))}
+         |> put_flash(:info, "Product purchased successfully!")
+         |> push_redirect(to: socket.assigns.return_to)}
 
-      {:error, %Ecto.Changeset{} = changeset} ->
-        {:noreply, assign(socket, :changeset, changeset)}
+      {:error, message} ->
+        {:noreply,
+         socket
+         |> put_flash(:error, message)
+         |> push_redirect(to: socket.assigns.return_to)}
     end
   end
+
+  defp get_available_sizes(product) do
```

Comment:
This is out of context, but is basically to only display available product sizes